src/logplex_logs_rest.erl

```diff
@@ -259,3 +288,14 @@ respond(Code, Text, Req, State) ->
     Req3 = cowboy_req:set_resp_body(Text, Req2),
     {ok, Req4} = cowboy_req:reply(<<(integer_to_binary(Code))/binary, " ", Text/binary>>, Req3),
     {halt, Req4, State}.
+
+decompress_body(CompressedBody, ZPort) ->
+    try
+        [Inflated] = zlib:inflate(ZPort, CompressedBody),
+        %% reset rather than end so we can call this function again if
+        %% need be.
+        ok = zlib:inflateReset(ZPort),
+        {ok, Inflated}
+    catch _Class:Reason ->
+            {error, Reason}
```

Comment:
You know, I am not sure why I did it this way, since there will be one of these per request.  I think that you're right that the right thing to do is just to call `zlib:gunzip/1` here.